airflow/api_connexion/endpoints/version_endpoint.py

```diff
@@ -32,6 +33,7 @@ class VersionInfo(NamedTuple):
     git_version: Optional[str]
 
 
+@security.requires_access([("can_read", "Version")])
 def get_version():
```

Comment:
This will allow the client's behavior to be changed before authorization is performed.  I am able to imagine that other authorization methods may be required depending on the version.

Kubernetes API don't have aaccesss control on /version endpoint
```
$ ENDPOINT=$(gcloud container clusters describe airflow-cluster --zone us-central1 --format='value(endpoint)')
$ curl --insecure https://$ENDPOINT/version
{
  "major": "1",
  "minor": "16+",
  "gitVersion": "v1.16.13-gke.1",
  "gitCommit": "688c6543aa4b285355723f100302d80431e411cc",
  "gitTreeState": "clean",
  "buildDate": "2020-07-21T02:37:26Z",
  "goVersion": "go1.13.9b4",
  "compiler": "gc",
  "platform": "linux/amd64"
}
```